src/python/pants/backend/go/build.py

```diff
@@ -57,24 +53,6 @@
 logger = logging.getLogger(__name__)
 
 
-class GoBuildSubsystem(GoalSubsystem):
```

Comment:
This probably needs to support `check` now. Using `package` only works if your project produces binaries, which a pure library does not.

Reply:
And unlike, let's say Rust, Go libraries are consumed via source, not binary artifacts, so there is generally no binary library artifact to build.

Reply:
I see. I'll add back for now.

Reply:
To refine my point, Rust sources are also what library users do use, but `cargo build` will at least build a crate library's binary artifact since that notion exists. `go build` just goes package by package to build but there is nothing really tying all of the packages in an external module together into one binary artifact.